src/parse/decl.rs

```diff
@@ -2073,20 +2127,35 @@ mod tests {
             // possibly with trailing commas
             parse("int a[] = {1, 2, 3,};"),
             parse("int a[3] = {1, 2, 3,};"),
+            // or nested
+            parse("int a[][3] = {{1, 2, 3}};"),
+            parse("int a[][3] = {{1, 2, 3,}};"),
+            parse("int a[][3] = {{1, 2, 3,},};"),
+            parse("int a[1][3] = {{1, 2, 3,},};"),
+            // with misleading braces
+            parse("int a[][3] = {1, 2, 3};"),
+            parse("int a[][3] = {1, 2, 3,};"),
         ];
         for res in &parsed {
-            let matches = match res {
-                Some(Ok(Locatable {
+            match res.as_ref().unwrap() {
+                Ok(Locatable {
                     data:
                         Declaration {
                             init: Some(Initializer::InitializerList(_)),
                             ..
                         },
                     ..
-                })) => true,
-                _ => false,
+                }) => {}
+                Ok(Locatable { data, .. }) => {
+                    panic!("expected initializer list, got declaration: {}", data)
+                }
+                Err(Locatable { data, .. }) => {
```

Comment:
This could just be `Err(error) => {` so that you keep the location information in the panic message.

I just realized that `CompileError` does not implement display, so you can disregard this nit. Not in this PR, but should we move the code for printing errors into an implementation of `Display` on `CompileError` so it can be used elsewhere? (I think this is currently in the `pritty_print` function in main.rs)

Reply:
We can't move pretty-printing into `CompileError` because it requires having a handle to the file source, not just the current error. Otherwise we can't print the line of code or calculate the line/column.

Reply:
Also, I specifically got rid of `location` because I don't care about it for unit tests.